test/powershell/Modules/Microsoft.PowerShell.Utility/Invoke-Item.Tests.ps1

```diff
@@ -37,14 +37,52 @@ Describe "Invoke-Item" -Tags "CI" {
         $testfile = Join-Path $TestDrive testfile.txt
     }
 
-    It "Should invoke a text file without error" -Skip:($IsWindows -and $IsCoreCLR) {
+    It "Should invoke a text file without error on non-Windows" -Skip:($IsWindows) {
         $debugfn = NewProcessStartInfo "-noprofile ""``Invoke-Item $testfile`n" -RedirectStdIn
         $process = RunPowerShell $debugfn
         EnsureChildHasExited $process
         $process.ExitCode | Should Be 0
     }
 
+<#
     It "Should throw not supported on Windows with .NET Core" -Skip:($IsLinux -or $IsOSX -or !$IsCoreCLR) {
         { Invoke-Item $testfile }| Should Throw "Operation is not supported on this platform."
     }
+#>
 }
+
+Describe "Invoke-Item deterministic tests on Windows" -Tags "CI","RequireAdminOnWindows" {
+    BeforeAll {
+        $testfilename = "testfile.!!testext!!"
+        $testfilepath = Join-Path $TestDrive $testfilename
+        $renamedtestfilename = "renamedtestfile.!!testext!!"
+        $renamedtestfilepath = Join-Path $TestDrive $renamedtestfilename
+        remove-item $testfilepath -ErrorAction SilentlyContinue
+        remove-item $renamedtestfilepath -ErrorAction SilentlyContinue
+        new-item $testfilepath | Out-Null
+
+    }
+
+    AfterAll {
+        remove-item $testfilepath -ErrorAction SilentlyContinue
+        remove-item $renamedtestfilepath -ErrorAction SilentlyContinue
+
+    }
+
+    It "Should invoke a file without error on Windows w/o .NET Core" -Skip:(-not $IsWindows -or ($IsWindows -and $IsCoreCLR)) {
+        cmd.exe /c assoc .!!testext!!=!!testext!!.FileType
+        cmd.exe /c ftype !!testext!!.FileType=cmd.exe /c rename $testfilepath $renamedtestfilename
+        invoke-item $testfilepath
+        $PSVersionTable | Out-File c:\1\1.txt
+        # Wiaiting subprocess start and rename file
+        Start-Sleep -Milliseconds 500
```

Comment:
This is not very reliable, we often see problem in CI with timeouts, because it's slower. Please, consider replacing it by something like

```
startTime = now
while (renaming didn't happen)
{
  sleep -Milliseconds  100
  if (now  -  startime > 5 seconds) { throw timeout }
}
```